### DISCUSSION TITLE
One-click Blocking

### DISCUSSION THREAD

**Discussion by ghost** [AprilSylph/XKit-Rewritten#732] [Suggestion Box]
Is this sort of functionality possible in X-kit Rewritten or would it need to be it's own extension, like 'Filtering+'?

**Comment by marcustyphoon** [comment]
Well, what exactly are you envisioning? Filtering+ uses an API that allows it to add functionality to the right click menu when you select text. One wouldn't need that to add a button to the blog cards that pop up when you mouse over a user's name, for example. But don't those have a block button in the menu anyway? (Can't check at the moment.)

**Reply by ghost** [reply]
I was a little bit vague lol To expand on my first comment, I was just curious if X-kit had the ability to skip additional prompts that make the process a bit long in the tooth, at least for repeat. What with needing to open the pop-up menu, click block, confirm your choice, and then confirm once more for extra measure. Makes it really monotonous. 

It would definitely be nice to simplify it to a context menu click like with Filtering+, but that's probably a bit out of the scope of this project.

Also wrt the blog card pop-up, it does also have a block button, and afaik acts the same.

**Reply by marcustyphoon** [reply]
Yeah; while I don't think we *would* do either:

- As far as I know, Filtering+ doesn't from a technical standpoint *have* to be a separate extension from XKit Rewritten to use the right click menu API, though I think there are browser-related reasons why it would be Kind Of Annoying:tm:.
- If one really did want to make a "help block people as fast as possible" script I don't think the right click menu is technically necessary. Adding a button to the blog card pop-up that blocks the person without a confirmation prompt is possible (though there isn't really any room for it). Or adding a menu item to the meatballs menu in the top right of the post that's just "Block (skip confirmation). All of these would basically be two-click blocks instead of the current three-click (if you count hovering over the blog card as a click). And in the extreme case, one could replace the "follow" button that appears next to blog names with a "block" button that would flip over to "blocked!" when clicked (and flip back if clicked again); that would really be "one-click blocking."

**Reply by ghost** [reply]
Mmm, fair. Interesting to conceptualize at least lol Thanks for the clarification, I appreciate it!
